raptor_dbw_can/src/DbwNode.cpp

```diff
@@ -848,9 +851,46 @@ void DbwNode::recvCAN(const can_msgs::msg::Frame::SharedPtr msg) {
                 }
             } break;
         }
+
+        // generates TireTemp messages and assign them to TireTempReport.
+        tire_temp_report_msg.front_left = DbwNode::generateTireTemp(fl_tire_temperature);
+        tire_temp_report_msg.front_right = DbwNode::generateTireTemp(fr_tire_temperature);
+        tire_temp_report_msg.rear_left = DbwNode::generateTireTemp(rl_tire_temperature);
+        tire_temp_report_msg.rear_right = DbwNode::generateTireTemp(rr_tire_temperature);
     }
 }
 
+// computes values of TireTemp message for a given array of tire temperatures.
+deep_orange_msgs::msg::TireTemp DbwNode::generateTireTemp(float32[] tire_temperature) {
+    deep_orange_msgs::msg::TireTemp tireTemp;
+    float32 sum = 0.0;
+    int count = 0;
+    float32[] temps;
+
+    for (int i = 0; i < 16; i++) {
+        float32 curr_temp = tire_temperature[i];
+        if (curr_temp >= 0.0) {
+            sum += curr_temp;
+            count += 1;
+            temps.push_back(curr_temp);
+        }
+    }
+
+    sort(temps, temps + count);
+
+    if (count % 2 != 0) {
+        tireTemp.median = (float32)temps[count/2];
```

Comment:
please use `static_cast<float>` instead of these c-style casts. 

see https://stackoverflow.com/questions/103512/why-use-static-castintx-instead-of-intx for why we use this